consensus/spos/bls/subroundEndRound.go

```diff
@@ -155,6 +156,94 @@ func (sr *subroundEndRound) isBlockHeaderFinalInfoValid(cnsDta *consensus.Messag
 	return true
 }
 
+// receivedInvalidSignersInfo method is called when a message with invalid signers has been received
+func (sr *subroundEndRound) receivedInvalidSignersInfo(_ context.Context, cnsDta *consensus.Message) bool {
```

Comment:
@SebastianMarian is there a check for equality somewhere between the header hash for this consensus round and the header hash in this specific message? can't find exactly where this is done.

Reply:
Yes it is, In doEndRoundJobByParticipant -> doEndRoundJobByParticipant -> isConsensusHeaderReceived
The validator will make a hash on its resulted processed header and after this it will set nil on leaderSignature, pubKeyBitmap and signature fields on the received header(s) from the proposer and will check the equality between them. If it will find one that matches with its own, this method will return true alongside the original received header, with all these three fields set, and afterwards it will proceed to commit it .

```
func (sr *subroundEndRound) isConsensusHeaderReceived() (bool, data.HeaderHandler) {
	if check.IfNil(sr.Header) {
		return false, nil
	}

	consensusHeaderHash, err := core.CalculateHash(sr.Marshalizer(), sr.Hasher(), sr.Header)
	if err != nil {
		log.Debug("isConsensusHeaderReceived: calculate consensus header hash", "error", err.Error())
		return false, nil
	}

	receivedHeaders := sr.GetReceivedHeaders()

	var receivedHeaderHash []byte
	for index := range receivedHeaders {
		receivedHeader := receivedHeaders[index].ShallowClone()
		err = receivedHeader.SetLeaderSignature(nil)
		if err != nil {
			log.Debug("isConsensusHeaderReceived - SetLeaderSignature", "error", err.Error())
			return false, nil
		}

		err = receivedHeader.SetPubKeysBitmap(nil)
		if err != nil {
			log.Debug("isConsensusHeaderReceived - SetPubKeysBitmap", "error", err.Error())
			return false, nil
		}

		err = receivedHeader.SetSignature(nil)
		if err != nil {
			log.Debug("isConsensusHeaderReceived - SetSignature", "error", err.Error())
			return false, nil
		}

		receivedHeaderHash, err = core.CalculateHash(sr.Marshalizer(), sr.Hasher(), receivedHeader)
		if err != nil {
			log.Debug("isConsensusHeaderReceived: calculate received header hash", "error", err.Error())
			return false, nil
		}

		if bytes.Equal(receivedHeaderHash, consensusHeaderHash) {
			return true, receivedHeaders[index]
		}
	}

	return false, nil
}

```

Reply:
my question was about the equality between this consensus message header hash and the header hash for the current consensus. I see you suggested below some added conditions that would fix this.

Reply:
Yes, that's right, this check is missing, and I also suggested it below the other missing conditions which should be added.